viral/cache_2p_sessions.py

```diff
@@ -147,9 +147,77 @@ def add_daq_times_to_trial(
         )
 
 
+def extract_frozen_wheel_chunks(
+    stack_lengths_tiffs: np.ndarray,
+    valid_frame_times: np.ndarray,
+    behaviour_times: np.ndarray,
+    sampling_rate: int,
+) -> tuple[tuple[int, int], tuple[int, int]]:
+    """Extract start and end frame index for pre-training and post-training imaging chunks.
+    Args:
+        stack_lengths_tiffs (np.ndarray):           A NumPy array of length tiffs, with the number of frames in each tiff.
+        valid_frame_times (np.ndarray):             A NumPy array with times for valid frames.
+        behaviour_times (np.ndarray):               A NumPy array with all times when a Bpod spacer signal occured.
+        sampling_rate (int):                        The sampling rate of the DAQ system in Hz.
+
+    Returns:
+        tuple[tuple[int, int], tuple[int, int]]:    First chunk and last chunk, with their respective start and end frame.
+    """
+
+    # first chunk (before behavioural chunks)
+    first_chunk_len = stack_lengths_tiffs[0]
+    first_chunk = (0, first_chunk_len - 1)  # start and end frame
+    first_chunk_times = valid_frame_times[first_chunk[0] : first_chunk[1] + 1]
```

Comment:
Sorry I know this is gonna change all the code, but i think we shouldn't offset everything by 1.

Currently if you want to get the frozen wheel chunk from the result returned here you need to do

```python
dff[first_chunk[0]: first_chunk[1] + 1]
```

I would just stick to everything being 0 indexed. I know that you're correct currently e.g. the 36999th frame is the final one but i think this is easier going forward